lib/teleterm/clusters/storage.go

```diff
@@ -218,6 +219,8 @@ func (s *Storage) fromProfile(profileName, leafClusterName string) (*Cluster, *c
 		return nil, nil, trace.Wrap(err)
 	}
 
+	s.CustomHardwareKeyPrompt = hardwarekey.NewPINCachingPrompt(s.CustomHardwareKeyPrompt, cfg.PIVPINCacheTTL)
```

Comment:
Regarding data race from https://github.com/gravitational/teleport/issues/29771#issuecomment-2814505436: what's the idea behind setting `s.CustomHardwareKeyPrompt` here? `fromProfile` is called on almost every RPC in tshd.

Reply:
I was under the impression that the client would be cached so it would only be called once and not concurrently, my mistake. I was trying to set the `pin_cache_ttl` from wherever we load it from the profile or ping response.

Reply:
Yeah it's kind of a mess because we do cache the client, but then there's `lib/teleterm/clusters.Cluster` which implements a bunch of business logic and you can get it only through `clusters.Storage`.